### ISSUE TITLE
grid optimization for broadening parameters

### ISSUE THREAD

**Issue by HajimeKawahara** [HajimeKawahara/exojax#366]
I postponed this issue when releasing v1.3. 
Related to #365,  we realized that the appropriate grid size of broadening parameters was important for real life ;)

- should be considered with #327.

**Comment by HajimeKawahara** [comment]
As discussed with @ykawashima, we decided to make linear grids of broadening parameters instead of log.

**Comment by HajimeKawahara** [comment]
Memo: Two equations to think about the errors of the current implementation

![image](https://user-images.githubusercontent.com/15956904/229380192-eb60f8f4-a05b-4248-b18e-a99c7978b4f5.png)

![image](https://user-images.githubusercontent.com/15956904/229380228-61370c22-a6c3-4586-a18c-3ec4ff79f749.png)

A potential improvement is we change `Tref` for n. For instance, when our range is 400-2000K, the current implementation requires, np.log(2000.)-np.log(273.) = 2 of the boost factor. but if setting Tref at the middle of log 400 and 2000 K, 

```python
>>> tmid=(np.log(2000.)+np.log(400.))/2
>>> np.log(2000.)-tmid
0.8047189562170498
>>> (np.log(2000.)-np.log(273.))/(np.log(2000.)-tmid)
2.474690883036675
```
we get about 2.5 times smaller boost factor, which reduces the device memory use.

**Comment by HajimeKawahara** [comment]
The 2D C should be 4/3 times smaller than the 1D case:

![diff2D1D_broadening](https://user-images.githubusercontent.com/15956904/229382931-4de419e1-cd3f-4d61-b94b-cadbd8f73b43.png)

**Comment by HajimeKawahara** [comment]
example: `test_broadpar_grid_as_a_function_of_Tref_broadening(db)` in `tests/unittests/spec/premodit/premodit_test.py`

**Comment by HajimeKawahara** [comment]
N_gamma, N_n

|     |current |this  |
|-----|---------|------|
| H2O | 21, 17|28, 8|
| CH4 | 5, 13  |12, 7|
| CO|5, 2     |5, 2  |
| NH3|8, 6     |13, 3|
| H2S|9, 2     |12, 2|

**Comment by HajimeKawahara** [comment]
implemented #370
